### PR TITLE
[DOCS] Add token and HTTPS requirements for Kerberos

### LINKED ISSUE #54621: Kerberos Docs should state that HTTPS is required (for tokens)
Doc: https://www.elastic.co/guide/en/elasticsearch/reference/7.6/kerberos-realm.html

Like SAML and OIDC, Kerberos auth requires that the token service is enabled.

In prod mode, the token service is tied to `xpack.security.http.ssl.enabled` - it is automatically turned on when HTTPS is turned on, and bootstrap checks prevent the token service from being enabled when HTTPS is off.

In effect, this means that Keberos requires https (`xpack.security.http.ssl.enabled: true`), but we do not explicitly mention that in the Kerberos realm docs.

### PR DESCRIPTION
**Pull request by lcawl**
Fixes https://github.com/elastic/elasticsearch/issues/54621

In particular, it re-uses content from the SAML docs (https://www.elastic.co/guide/en/elasticsearch/reference/current/saml-guide-authentication.html steps about enabling TLS for HTTP and enabling the token service) in the Kerberos docs.

Preview:
* https://elasticsearch_57180.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/kerberos-realm.html#kerberos-realm-configuration

### CODE REVIEW

**Comment by tvernum** [comment]
My brain was only half working at the time I opened the linked issue, and I now realise that the require to have HTTPs and the token service is only true for Kerberos with Kibana, but it is not required if you use Kerberos directly against Elasticsearch.

I'll put some specific suggestions about how we might try and make that clear.

File: x-pack/docs/en/security/authentication/configuring-kerberos-realm.asciidoc [hunk 1]
```diff
@@ -46,6 +46,28 @@ For more information on Java GSS, see
 https://docs.oracle.com/javase/10/security/kerberos-requirements1.htm[Java GSS Kerberos requirements]
 --
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/configuring-kerberos-realm.asciidoc:48]: The steps below are required if you want to use Kerberos Authentication via Kibana (which most people will want, but maybe not all).

I think we might want an introductory note here, but it's hard to say it refers to "the two steps below, but not the subsequent section ...".

File: x-pack/docs/en/security/authentication/configuring-kerberos-realm.asciidoc [hunk 2]
```diff
@@ -46,6 +46,28 @@ For more information on Java GSS, see
...
 
+. Enable TLS for HTTP.
++
+--
+If your {es} cluster is operating in production mode, you must configure the
+HTTP interface to use SSL/TLS before you can enable Kerberos authentication. For
+more information, see <<tls-http>>.
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/configuring-kerberos-realm.asciidoc:54]: Maybe it's as simple as just having:

```suggestion
more information, see <<tls-http>>.

This step is necessary to support Kerberos authentication via {kib}.
It is not required for Kerberos authentication directly against the {es} Rest API.
```

File: x-pack/docs/en/security/authentication/configuring-kerberos-realm.asciidoc [hunk 3]
```diff
@@ -46,6 +46,28 @@ For more information on Java GSS, see
...
+--
+The {es} Kerberos implementation makes use of the {es} token service. If you
+configure TLS on the HTTP interface, this service is automatically enabled. It
+can be explicitly configured by adding the following setting in your
+`elasticsearch.yml` file:
+
+[source, yaml]
+------------------------------------------------------------
+xpack.security.authc.token.enabled: true
+------------------------------------------------------------
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/configuring-kerberos-realm.asciidoc:68]:
```suggestion
------------------------------------------------------------

This step is necessary to support Kerberos authentication via {kib}.
It is not required for Kerberos authentication directly against the {es} Rest API.
```

File: x-pack/docs/en/security/authentication/configuring-kerberos-realm.asciidoc (hunk 1 above)
Comment (lcawl) [reply x-pack/docs/en/security/authentication/configuring-kerberos-realm.asciidoc:48]: Alternatively, we could have indented them both under a "If you want to use Kerberos with Kibana:" step, but I think your suggestions are just as clear. Thanks!

**Comment by lcawl** [comment]
@elasticmachine update branch

**Pull request merged**